pkg/api/aim/query/query_test.go

```diff
@@ -110,24 +110,21 @@ func (s *QueryTestSuite) TestPostgresDialector_Ok() {
 			name:  "TestNegativeInteger",
 			query: `run.metrics['my_metric'].last < -1`,
 			expectedSQL: `SELECT "run_uuid" FROM "runs" ` +
-				`LEFT JOIN latest_metrics metrics_0 ON runs.run_uuid = metrics_0.run_uuid AND metrics_0.key = $1 ` +
-				`WHERE ("metrics_0"."value" < $2 AND "runs"."lifecycle_stage" <> $3)`,
+				`WHERE ("latest_metrics"."key" = $1 AND ("latest_metrics"."value" < $2 AND "runs"."lifecycle_stage" <> $3))`,
```

Comment:
I am concerned that this test was modified. This query does not look equivalent to me, and I imagine that something like `run.metrics['metric1'].last < -1 and run.metrics['metric2'] >= -1` will not return what it should.

Reply:
I think we need to move the metric key clause from the JOIN to the WHERE (so that additional conditions could use the same JOIN) but I think it is is equivalent. The JOIN is handled outside of the QueryParser now (since columns are also referenced there), so they do not appear in this unit test.

Reply:
I think that having a single JOIN for all the `latest_metrics` won't allow you to match runs with something like `run.metrics['metric1'].last < -1 and run.metrics['metric2'] >= -1`, as your single run row will not contain the information about both metrics at the same time.

Reply:
to be clear, this syntax returns only runs that have both `metric1` and `metric2`. I think it's working as expected now.